packages/graphql/lib/src/link/web_socket/link_web_socket.dart

```diff
@@ -41,15 +52,20 @@ class WebSocketLink extends Link {
   }
 
   /// Connects or reconnects to the server with the specified headers.
-  void connectOrReconnect() {
-    _socketClient?.dispose();
+  void connectOrReconnect() async {
```

Comment:
When I run the starwars example on this branch I get
> The method 'subscribe' was called on null.

I believe this is because this makes it so we need `_doOperation` to be `async`. Not sure what the right approach is though - don't have much experience with stream operations. I think we basically want a stream that adds waiting for the future to the first item.

Reply:
I haven't used the library for a while now, so it's not that fresh in mind right now unfortunately.

But I believe that I have added `onInitResponse` which you can listen to (may be by using `StreamBuilder` or in a stateful widget) that will allow you to find-out if you're ready to send subscription or not.

In fact, along with the primary goal of trying to know if the socket is connected or not, `onInitResponse` will allow to know whether we can actually start to send requests or not. Even if Websocket is connected, it doesn't mean that you can actually send a request, you have to be fully initialized before starting to send requests.

Otherwise, are you suggesting that we revert back to non-async `connectOrReconnect()` ?